static/TODO.go

```diff
@@ -0,0 +1,28 @@
+// +build !assets
```

Comment:
Now that the UI & swagger are getting built externally, do we need to keep this `assets` flag & the `TODO` case? It seems to be a frequent tripping point for users trying to build from source by using `go build` instead of `make`. Could we instead have the go-build always assume assets are available locally, and raise a build-time error if they're not present?

Reply:
Based on my attempts, the flag and `TODO` case are really needed so that the test suite can be run without getting a build error. Things like the `golint` and `gotest` CI jobs which run tests without necessarily traversing through & running the `go:generate` here would fail otherwise since the functions in the `TODO` file don't exist. I do feel like there might be some way to get around this that I'm missing though.

One of the other benefits of `go:embed` is that - after a fashion - if the embedding assets aren't generated (like for testing), the tests can still run and the binary can even compile, but we could just send a 404 if one of the assets is requested.

So my position is that: It's quite possible that I'm missing a better way to handle this currently...but if I'm not, I'd punt cleaning up the `assets` flag until/if we get around to incorporating asset compression and using `go:embed` after all.

Reply:
I looked for other ways to work around the ability to run tests without pulling in the external assets and couldn't come up with anything. Packages other than `go-bindata` also seem to rely on using some kind of `go:generate` to generate some code that you reference via function or import as a package etc., so it has to exist for tests to be able to run. Build tags seem like the cleanest way to say "under this set of circumstances, use this function...and use a different one under a different set of circumstances". I guess the other option might be to generate all of the assets even for running the unit tests etc., but that seems not ideal for other reasons.